internal/policy/operator/restricted_network_aware.go

```diff
@@ -0,0 +1,99 @@
+package operator
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/api/pkg/manifests"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/check"
+	libcsv "github.com/redhat-openshift-ecosystem/openshift-preflight/internal/csv"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/image"
+)
+
+var _ check.Check = &FollowsRestrictedNetworkEnablementGuidelines{}
+
+type FollowsRestrictedNetworkEnablementGuidelines struct{}
+
+func (p FollowsRestrictedNetworkEnablementGuidelines) Validate(ctx context.Context, imgRef image.ImageReference) (bool, error) {
+	return p.validate(ctx, imgRef.ImageFSPath)
+}
+
+//nolint:unparam // ctx is unused. Keep for future use.
+func (p FollowsRestrictedNetworkEnablementGuidelines) getBundleCSV(ctx context.Context, bundlepath string) (*operatorsv1alpha1.ClusterServiceVersion, error) {
+	bundle, err := manifests.GetBundleFromDir(bundlepath)
+	if err != nil {
+		return nil, err
+	}
+	return bundle.CSV, nil
+}
+
+func (p FollowsRestrictedNetworkEnablementGuidelines) validate(ctx context.Context, bundledir string) (bool, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	csv, err := p.getBundleCSV(ctx, bundledir)
+	if err != nil {
+		return false, err
+	}
+
+	// If the CSV does not claim to support disconnected environments, there's no reason to check that it followed guidelines.
+	if !libcsv.HasInfrastructureFeaturesAnnotation(csv) {
+		logger.Info("warning: this operator does not have the infrastructure-features annotation, unable to determine if it supports disconnected environments.")
+		return false, nil
+	}
+
+	if !libcsv.SupportsDisconnected(csv.Annotations[libcsv.InfrastructureFeaturesAnnotation]) {
+		logger.Info("warning: the infrastructure-features enabled for this operator did not include the \"disconnected\" identifier.")
```

Comment:
Why is this a warning? If I'm not expecting to have a disconnected operator, I'd be confused by this. I'd be ok with the log message, just don't think it should be specified as "warning:"

Reply:
That's fair. Not sure what I was thinking re: the warning. Thinking it may have been a copy-pasta. 

Now that I think about it, should I log this at all? :thinking:

Reply:
I think it should be logged. Just not as "warning:"..

Let's think this through. As a user of a normal environment, if I see this message, what do I think? "Am I supposed to have 'disconnected' there?"

As a disconnected user, if I *don't* see this message, will I know that it's not working? 

I think the compromise might be this:

the infrastructure-features enabled for this operator did not include the "disconnected" identifier. If the operator is not targeted for a disconnected environment, one can safely ignore this message.

Reply:
I had similar thoughts on all of this logic as well (and commented on the commit and not the PR, no idea where that is at anymore), we are catering to the few, to confuses the masses.

Reply:
@acornett21 I totally get where you are coming from there. I think the important thing here is to make sure that we fail as fast as possible, and communicate to those that don't care about it... "You can ignore this". So, we just have to make sure that the first check is the one that fails the fastest. 

Also, it might make sense to bookend these messages. Log at the beginning of the function "You can ignore this if you are not a disconnected operator", and defer logger.Info("Ignoring complete")

Reply:
@bcrochet there's an initial phase that checks that the operator has the annotation and that it contains the expected value. We don't actually check for readiness unless you pass both of those checks.

To that end, I've added "safe to ignore" to both of their log lines, and removed all the `warning:` text in the latest force-push https://github.com/redhat-openshift-ecosystem/openshift-preflight/commit/2e4f1be46c633407da33dfef9325245895f35078

Reply:
@bcrochet All that makes sense and I think @komish is doing that, my comment was more about the verbiage specifically. I think the latest commit clears/cleans things up nicely.